implementations/rust/ockam/ockam_node/src/executor.rs

```diff
@@ -113,6 +115,21 @@ impl Executor {
         Ok(res)
     }
 
+    async fn wrapper<F, T>(sender: SmallSender<NodeMessage>, future: F) -> Result<T>
+    where
+        F: Future<Output = Result<T>> + Send + 'static,
+        // T: Sync + Send + 'static,
+    {
+        match future.await {
+            Ok(val) => Ok(val),
+            Err(e) => {
+                // let sender = self.sender();
+                let _ = sender.send(NodeMessage::AbortNode).await;
```

Comment:
It looks like `AbortNode` is not handled properly and doesn't really stop a node (should be a bug, will create an issue for that).
However, you can find the correct way to stop a node inside `Context::stop` function.